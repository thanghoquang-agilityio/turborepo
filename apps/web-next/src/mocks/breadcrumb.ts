export const MOCK_BREADCRUMB_ITEMS = [
  { name: 'Products', link: 'products' },
  { name: 'Detail Product', link: '7' },
]
