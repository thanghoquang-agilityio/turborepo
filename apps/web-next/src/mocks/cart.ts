import { CartItemResponse, LegacyCartItemResponse } from '@/types'
import { MOCK_PRODUCT_VARIANTS } from './product'

export const MOCK_CART_ITEM = {
  id: '1',
  variantId: '1',
  name: 'Embrace Sideboard',
  color: 'wheat',
  size: 'small',
  quantityDefault: 2,
  stock: 4,
  price: 53,
  star: 1,
  comments: 1,
  tax: 1,
  discount: 1,
}

// New flattened cart response structure
export const MOCK_CART_RESPONSE: CartItemResponse[] = [
  {
    id: 1,
    documentId: 'cart-item-1',
    size: 'small',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[0],
    userId: 1,
  },
  {
    id: 2,
    documentId: 'cart-item-2',
    size: 'large',
    quantity: 2,
    createdAt: '2024-01-01T00:00:00.000Z',
    updatedAt: '2024-01-01T00:00:00.000Z',
    publishedAt: '2024-01-01T00:00:00.000Z',
    productVariantId: MOCK_PRODUCT_VARIANTS[1],
    userId: 1,
  },
]

// Legacy cart response for backward compatibility
export const MOCK_CART_RESPONSE_LEGACY: LegacyCartItemResponse[] = [
  {
    id: '1',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '2,10,0,23,20',
            tax: 12,
          },
        },
      },
      size: 'small',
      quantity: 2,
    },
  },
  {
    id: '2',
    attributes: {
      productVariantId: {
        data: {
          id: '1',
          attributes: {
            size: 'small,medium,large,extra large,XXL',
            color: 'slate blue',
            stock: '20,10,20,23,20',
            tax: 12,
          },
        },
      },
      size: 'large',
      quantity: 2,
    },
  },
]

export const MOCK_ORDER_SUMMARY = {
  subTotal: 396,
  itemTotal: 3,
}
