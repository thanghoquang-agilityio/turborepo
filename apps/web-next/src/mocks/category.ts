export const MOCK_CATEGORY_LIST = [
  {
    name: 'Kid',
    value: 'kid',
    total: 18,
  },
  {
    name: 'Man',
    value: 'man',
    total: 12,
  },
  {
    name: 'Woman',
    value: 'woman',
    total: 23,
  },
  {
    name: 'Sport',
    value: 'sport',
    total: 34,
  },
]

export const MOCK_CATEGORY_RESPONSE = [
  {
    name: 'Kid',
    value: 'kid',
    products: {
      data: [],
    },
  },
  {
    name: 'Man',
    value: 'man',
    products: {
      data: [],
    },
  },
  {
    name: 'Woman',
    value: 'woman',
    products: {
      data: [],
    },
  },
  {
    name: 'Sport',
    value: 'sport',
    products: {
      data: [],
    },
  },
]
