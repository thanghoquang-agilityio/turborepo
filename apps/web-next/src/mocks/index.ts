export * from './breadcrumb'
export * from './cart'
export * from './category'
export * from './image'
export * from './product'
export * from './radio-group'
export * from './user'
