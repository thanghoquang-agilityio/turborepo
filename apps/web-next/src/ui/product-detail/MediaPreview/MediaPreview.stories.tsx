import type { Meta, StoryObj } from '@storybook/react'

import { IMAGE_DEFAULT } from '@/constants'

import MediaPreview from '.'

const meta = {
  title: 'Components/MediaPreview',
  tags: ['autodocs'],
  component: MediaPreview,
  parameters: {
    layout: 'centered',
  },
} satisfies Meta<typeof MediaPreview>

export default meta
type Story = StoryObj<typeof MediaPreview>

export const SingleImage: Story = {
  args: { images: [IMAGE_DEFAULT] },
}

export const MultipleImage: Story = {
  args: {
    images: [IMAGE_DEFAULT, IMAGE_DEFAULT, IMAGE_DEFAULT, IMAGE_DEFAULT],
  },
}
