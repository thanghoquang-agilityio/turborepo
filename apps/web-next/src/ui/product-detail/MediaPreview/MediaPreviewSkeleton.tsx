import { Skeleton } from '@repo/next-ui'

export const MediaPreviewImageSkeleton = () => (
  <div className='img-preview my-[30px] h-auto lg:my-0'>
    <div className='flex h-full flex-col items-center gap-4 lg:gap-[37px]'>
      <div className='flex h-[375px] w-full justify-center sm:h-[450px] md:h-[525px] lg:h-full lg:max-w-[498px] xl:max-w-[587px]'>
        <Skeleton className='bg-default-200 h-full w-full rounded-[17px]' />
      </div>
      <div className='hidden items-center sm:flex lg:gap-2.5'>
        <div className='sm:flex sm:gap-3 xl:gap-5'>
          {Array(4)
            .fill(0)
            .map((_, index) => (
              <Skeleton
                key={`product-image-${index}`}
                className='bg-default-200 flex items-center justify-center rounded-[14px] sm:h-[106px] sm:w-[106px] xl:h-[115px] xl:w-[115px]'
              />
            ))}
        </div>
      </div>
    </div>
  </div>
)
