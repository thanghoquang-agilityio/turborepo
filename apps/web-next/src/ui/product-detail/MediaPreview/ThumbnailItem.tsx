import { memo } from 'react'

// Component
import { CustomImage } from '@/components'
import { cn } from '@/lib/utils'
// Constants
import { API_IMAGE_URL } from '@/constants'

interface ThumbnailItemProps {
  onSelect: (id: number) => void
  id: number
  alt: string
  url: string
  isSelected?: boolean
}

export const ThumbnailItem = memo(
  ({ alt, url, isSelected, onSelect, id }: ThumbnailItemProps) => {
    return (
      <button
        data-testid={`thumbnail-item-${id}`}
        onClick={() => {
          onSelect(id)
        }}
        className={cn(
          'relative flex items-center justify-center rounded-[14px] sm:h-[106px] sm:w-[106px] xl:h-[115px] xl:w-[115px]',
          isSelected ? 'bg-linear-blue dark:bg-linear-white' : ''
        )}
      >
        <CustomImage
          alt={alt}
          src={`${API_IMAGE_URL}${url}`}
          className='h-full w-full rounded-[14px] p-[3px]'
          width={115}
          height={115}
          style={{ objectFit: 'cover' }}
        />
      </button>
    )
  }
)

ThumbnailItem.displayName = 'ThumbnailItem'
