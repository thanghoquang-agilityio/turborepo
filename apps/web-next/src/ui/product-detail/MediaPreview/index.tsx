'use client'

import { memo, useState } from 'react'

// Components
import { ButtonIcon, CustomImage } from '@/components'
// Constants
import { API_IMAGE_URL } from '@/constants'
import { ArrowLeftIcon, ArrowRightIcon } from '@/icons'
// Types
import { ImageModel } from '@/types'

import { ThumbnailItem } from './ThumbnailItem'

const MediaPreview = memo(({ images }: { images: ImageModel[] }) => {
  const [select, setSelect] = useState<number>(0)

  const handleSelect = (direction: 'left' | 'right') => {
    setSelect((prev) => {
      if (direction === 'left') return prev > 0 ? prev - 1 : images.length - 1

      return prev < images.length - 1 ? prev + 1 : 0
    })
  }

  const image = images[select]
  const { url = '', name = '' } = image

  return (
    <div className='flex h-full flex-col items-center gap-4 lg:gap-[37px]'>
      <div className='flex h-[375px] w-full justify-center overflow-hidden rounded-[17px] sm:h-[450px] md:h-[525px] lg:h-auto lg:max-w-[498px] xl:max-w-[587px]'>
        <CustomImage
          src={`${API_IMAGE_URL}${url}`}
          alt={name}
          className='h-[375px] rounded-[17px] sm:h-[450px] md:h-[525px] lg:h-[600px] xl:h-[691px]'
          width={587}
          height={691}
          priority
        />
      </div>
      {images.length > 1 && (
        <div className='flex items-center gap-4 sm:hidden lg:gap-2.5'>
          <ButtonIcon
            variant='ghost'
            size='sm'
            shape='circle'
            additionalClass='p-0 min-w-6 min-h-6 w-6 h-6 text-default'
            data-testid='btn-arrow-mobile-left'
            onClick={() => handleSelect('left')}
          >
            <ArrowLeftIcon />
          </ButtonIcon>
          <ButtonIcon
            variant='ghost'
            size='sm'
            shape='circle'
            additionalClass='p-0 min-w-6 min-h-6 w-6 h-6 text-default'
            data-testid='btn-arrow-mobile-right'
            onClick={() => handleSelect('right')}
          >
            <ArrowRightIcon />
          </ButtonIcon>
        </div>
      )}
      {images.length > 1 && (
        <div className='hidden items-center sm:flex lg:gap-2.5'>
          <ButtonIcon
            variant='ghost'
            size='sm'
            shape='circle'
            additionalClass='p-0 min-w-6 min-h-6 w-6 h-6 text-default xl:block lg:hidden'
            data-testid='btn-arrow-left'
            onClick={() => handleSelect('left')}
          >
            <ArrowLeftIcon />
          </ButtonIcon>
          <div className='sm:flex sm:gap-3 xl:gap-5'>
            {images.slice(0, 4).map(({ name, url }, index) => (
              <ThumbnailItem
                onSelect={setSelect}
                alt={name}
                url={url}
                key={index}
                id={index}
                isSelected={index === select}
              />
            ))}
          </div>
          <ButtonIcon
            variant='ghost'
            size='sm'
            shape='circle'
            additionalClass='p-0 min-w-6 min-h-6 w-6 h-6 text-default xl:block lg:hidden'
            data-testid='btn-arrow-right'
            onClick={() => handleSelect('right')}
          >
            <ArrowRightIcon />
          </ButtonIcon>
        </div>
      )}
    </div>
  )
})

MediaPreview.displayName = 'MediaPreview'
export default MediaPreview
