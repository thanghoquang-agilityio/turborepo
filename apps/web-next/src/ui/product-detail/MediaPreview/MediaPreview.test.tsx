import { fireEvent, render } from '@testing-library/react'

import { MOCK_LIST_IMAGE_RESPONSE } from '@/mocks'

import MediaPreview from '.'

describe('MediaPreview', () => {
  const renderComponent = () =>
    render(<MediaPreview images={MOCK_LIST_IMAGE_RESPONSE} />)

  it('should match snapshot', () => {
    const { container, getByTestId } = renderComponent()
    fireEvent.click(getByTestId('btn-arrow-left'))
    fireEvent.click(getByTestId('btn-arrow-mobile-left'))
    fireEvent.click(getByTestId('btn-arrow-right'))
    fireEvent.click(getByTestId('btn-arrow-mobile-right'))
    fireEvent.click(getByTestId('thumbnail-item-3'))

    expect(container).toMatchSnapshot()
  })
})
