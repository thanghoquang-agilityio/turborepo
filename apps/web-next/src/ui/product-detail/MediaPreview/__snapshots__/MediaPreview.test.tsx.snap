// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaPreview should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col gap-4 lg:gap-[37px] h-full items-center"
  >
    <div
      class="flex justify-center w-full xl:max-w-[587px] lg:max-w-[498px] lg:h-auto md:h-[525px] sm:h-[450px] h-[375px] rounded-[17px] overflow-hidden"
    >
      <img
        alt="default4"
        class="xl:h-[691px] lg:h-[600px] md:h-[525px] sm:h-[450px] h-[375px] rounded-[17px]"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="691"
        src="http://localhost/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=1200&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=640&q=75 1x, /_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=1200&q=75 2x"
        style="color: transparent; object-fit: cover;"
        width="587"
      />
    </div>
    <div
      class="flex sm:hidden lg:gap-2.5 items-center gap-4"
    >
      <button
        aria-label="Button Icon"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 border-none rounded-[100%] sm:min-w-[46px] sm:min-h-[46px] sm:p-[11px] p-0 min-w-6 min-h-6 w-6 h-6 text-default"
        data-testid="btn-arrow-mobile-left"
        type="button"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 4.5L8.25 12L15.75 19.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
        <span
          class="nextui-ripple"
          style="position: absolute; background-color: currentColor; border-radius: 100%; transform-origin: center; pointer-events: none; overflow: hidden; inset: 0; z-index: 0; top: 0px; left: 0px; width: 0px; height: 0px; transform: scale(0); opacity: 0.35; will-change: transform,opacity;"
        />
      </button>
      <button
        aria-label="Button Icon"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 border-none rounded-[100%] sm:min-w-[46px] sm:min-h-[46px] sm:p-[11px] p-0 min-w-6 min-h-6 w-6 h-6 text-default"
        data-focus="true"
        data-focus-visible="true"
        data-testid="btn-arrow-mobile-right"
        type="button"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5L15.75 12L8.25 19.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
        <span
          class="nextui-ripple"
          style="position: absolute; background-color: currentColor; border-radius: 100%; transform-origin: center; pointer-events: none; overflow: hidden; inset: 0; z-index: 0; top: 0px; left: 0px; width: 0px; height: 0px; transform: scale(0); opacity: 0.35; will-change: transform,opacity;"
        />
      </button>
    </div>
    <div
      class="hidden sm:flex lg:gap-2.5 items-center"
    >
      <button
        aria-label="Button Icon"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 border-none rounded-[100%] sm:min-w-[46px] sm:min-h-[46px] sm:p-[11px] p-0 min-w-6 min-h-6 w-6 h-6 text-default xl:block lg:hidden"
        data-testid="btn-arrow-left"
        type="button"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.75 4.5L8.25 12L15.75 19.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
        <span
          class="nextui-ripple"
          style="position: absolute; background-color: currentColor; border-radius: 100%; transform-origin: center; pointer-events: none; overflow: hidden; inset: 0; z-index: 0; top: 0px; left: 0px; width: 0px; height: 0px; transform: scale(0); opacity: 0.35; will-change: transform,opacity;"
        />
      </button>
      <div
        class="sm:flex xl:gap-5 sm:gap-3"
      >
        <button
          class="relative xl:h-[115px] xl:w-[115px] sm:h-[106px] sm:w-[106px] flex justify-center items-center rounded-[14px]"
          data-testid="thumbnail-item-0"
        >
          <img
            alt="default1"
            class="w-full h-full p-[3px] rounded-[14px]"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="115"
            src="http://localhost/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75 2x"
            style="color: transparent; object-fit: cover;"
            width="115"
          />
        </button>
        <button
          class="relative xl:h-[115px] xl:w-[115px] sm:h-[106px] sm:w-[106px] flex justify-center items-center rounded-[14px]"
          data-testid="thumbnail-item-1"
        >
          <img
            alt="default2"
            class="w-full h-full p-[3px] rounded-[14px]"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="115"
            src="http://localhost/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75 2x"
            style="color: transparent; object-fit: cover;"
            width="115"
          />
        </button>
        <button
          class="relative xl:h-[115px] xl:w-[115px] sm:h-[106px] sm:w-[106px] flex justify-center items-center rounded-[14px]"
          data-testid="thumbnail-item-2"
        >
          <img
            alt="default3"
            class="w-full h-full p-[3px] rounded-[14px]"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="115"
            src="http://localhost/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75 2x"
            style="color: transparent; object-fit: cover;"
            width="115"
          />
        </button>
        <button
          class="relative xl:h-[115px] xl:w-[115px] sm:h-[106px] sm:w-[106px] flex justify-center items-center rounded-[14px] bg-linear-blue dark:bg-linear-white"
          data-testid="thumbnail-item-3"
        >
          <img
            alt="default4"
            class="w-full h-full p-[3px] rounded-[14px]"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="115"
            src="http://localhost/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fnext-js-p6iu.onrender.com%2Fimages%2Fimage-not-available.webp&w=256&q=75 2x"
            style="color: transparent; object-fit: cover;"
            width="115"
          />
        </button>
      </div>
      <button
        aria-label="Button Icon"
        class="z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-small [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none data-[hover=true]:opacity-hover gap-2.5 font-semibold bg-background-100 border-none rounded-[100%] sm:min-w-[46px] sm:min-h-[46px] sm:p-[11px] p-0 min-w-6 min-h-6 w-6 h-6 text-default xl:block lg:hidden"
        data-testid="btn-arrow-right"
        type="button"
      >
        <svg
          class="w-full h-full"
          fill="none"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.25 4.5L15.75 12L8.25 19.5"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
        <span
          class="nextui-ripple"
          style="position: absolute; background-color: currentColor; border-radius: 100%; transform-origin: center; pointer-events: none; overflow: hidden; inset: 0; z-index: 0; top: 0px; left: 0px; width: 0px; height: 0px; transform: scale(0); opacity: 0.35; will-change: transform,opacity;"
        />
      </button>
    </div>
  </div>
</div>
`;
